controllers/capa_utils.go

```diff
@@ -0,0 +1,43 @@
+package controllers
+
+import (
+	"context"
+
+	capa "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3"
+	expcapa "sigs.k8s.io/cluster-api-provider-aws/exp/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func isRoleUsedElsewhere(ctx context.Context, ctrlClient client.Client, roleName string) (bool, error) {
+	var err error
+
+	var awsMachineTemplates capa.AWSMachineTemplateList
+	err = ctrlClient.List(
+		ctx,
+		&awsMachineTemplates,
+	)
+	if err != nil {
+		return false, err
+	}
+	for _, mt := range awsMachineTemplates.Items {
+		if mt.DeletionTimestamp != nil {
```

Comment:
Shouldn't we check for the role name? you only check for deletion timestamp Or I am blind and its somewhere here?

Reply:
🤦  I got distracted any totally forgot about the main thing. I'm an idiot

Reply:
Fixed